src/app/credentials.rs

```diff
@@ -1,51 +1,86 @@
 use secret_service::{EncryptionType, Error, SecretService};
-use serde::{Deserialize, Serialize};
-use serde_json::{from_str, to_string};
+use serde::{de::Error as _, ser::Error as _, Deserialize, Deserializer, Serialize, Serializer};
 use std::collections::HashMap;
-use std::str::from_utf8;
+use std::time::{Duration, SystemTime};
```

Comment:
maybe move these imports to the relevant module below, or even move this module to a file (optional)?